src/py/pyodide/_package_loader.py

```diff
@@ -149,6 +155,25 @@ def should_load_dynlib(path: str):
     return not PLATFORM_TAG_REGEX.match(tag)
 
 
+def set_wheel_installer(
+    filename: str,
+    archive: IO[bytes],
+    target_dir: Path,
+    installer: str | None,
+    source: str | None,
+):
+    z = ZipFile(archive)
+    name, version = filename.split("-", 2)[:-1]
+    dist_info_name = f"{name}-{version}.dist-info"
+    if not (zipfile.Path(z) / dist_info_name).is_dir():
+        raise Exception("archive is not a valid wheel")
+    dist_info = target_dir / dist_info_name
+    if installer:
+        (dist_info / "INSTALLER").write_text(installer)
+    if source:
+        (dist_info / "PYODIDE_SOURCE").write_text(source)
```

Comment:
I can't decide if we should be using [`direct_url.json`](https://packaging.python.org/en/latest/specifications/direct-url/#direct-url) for this if it's a separate use case.

Reply:
Ideally I think we should add `direct_url.json` according to the spec. This won't be sufficient for our use case, because if the package was NOT a `direct_url` then we want to know what index we downloaded it from. But we can do both.

Reply:
Well we could probably add custom fields if needed to that json

Reply:
There's also the `REQUESTED` file. I think in a subsequent PR we could add `direct_url` and `REQUESTED`.
It's kind of weird that the spec has nothing about recording which index the package came from.

Reply:
> Well we could probably add custom fields if needed to that json

We can add custom fields but according to the spec:

> This file MUST NOT be created when installing a distribution from an other type of requirement (i.e. name plus version specifier).

So that rules out using it to specify which index a package came from.